dotCMS/src/main/java/com/dotcms/rest/api/v1/temp/TempFileResource.java

```diff
@@ -185,6 +182,119 @@ private void saveMultipleBinary(final FormDataMultiPart body, final HttpServletR
         printResponseEntityViewResult(outputStream, objectMapper, completionService, futures);
     }
 
+    /**
+     *
+     * @param request
+     * @param response
+     * @param tempFileId
+     * @param body
+     * @return
+     */
+    @PUT
+    @Path("/id/{tempFileId: .*}")
+    @JSONP
+    @NoCache
+    @Produces("application/octet-stream")
+    @Consumes(MediaType.APPLICATION_JSON)
+    public final Response updateTempResource(@Context final HttpServletRequest request,
+                                             @Context final HttpServletResponse response,
+                                             @PathParam("tempFileId") final String tempFileId,
+                                             final PlainTextFileForm body) {
+        this.verifyTempResourceEnabled();
+        new WebResource.InitBuilder(request, response).requiredAnonAccess(AnonymousAccess.WRITE).rejectWhenNoUser(true).init();
```

Comment:
I am not sure if this service must be open to anonymous, the normal upload is ok, but this one is for very specific case, again not sure if it should or not @wezell ?

Reply:
Done!